### ISSUE TITLE
Some preset colors don't have enough contrast for WCAG

### ISSUE THREAD

**Issue by jonatanschroeder** [PrairieLearn/PrairieLearn#10855]
These are the colors used for assessment sets, tags, topics, sharing sets, question types, and fingerprints. They are defined in https://github.com/PrairieLearn/PrairieLearn/blob/master/apps/prairielearn/public/stylesheets/colors.css. Colors named *1 are generally fine, colors names *2 or *3 have a pretty low contrast.

Class name | Foreground | Background | Ratio | AA | AAA
--- | --- | --- | --- | --- | ---
.color-red1 | 000 | ffccbc | 14.6 | pass | pass
.color-red1.color-hover:hover | 000 | ffac9c | 11.6 | pass | pass
.color-red2 | fff | ff6c5c | 1.41 | fail | fail
.color-red2.color-hover:hover | fff | f75c4c | 1.23 | fail | fail
.color-red3 | fff | c72c1c | 1.39 | fail | fail
.color-red3.color-hover.color-hover:hover | fff | b71c0c | 1.67 | fail | fail
.color-pink1 | 000 | ffbcd8 | 13.4 | pass | pass
.color-pink1.color-hover:hover | 000 | ff8cc8 | 9.83 | pass | pass
.color-pink2 | fff | fa5c98 | 1.32 | fail | fail
.color-pink2.color-hover:hover | fff | ea4c88 | 1.10 | fail | fail
.color-pink3 | fff | ba1c58 | 1.56 | fail | fail
.color-pink3.color-hover:hover | fff | aa0c48 | 1.86 | fail | fail
.color-purple1 | 000 | dcc6e0 | 13.1 | pass | pass
.color-purple1.color-hover:hover | 000 | cea0e4 | 9.77 | pass | pass
.color-purple2 | fff | 9b59b6 | 1.18 | fail | fail
.color-purple2.color-hover:hover | fff | 8e44ad | 1.48 | fail | fail
.color-purple3 | fff | 5e147d | 2.84 | fail | fail
.color-purple3.color-hover:hover | fff | 4e046d | 3.40 | fail | fail
.color-blue1 | 000 | 39d5ff | 12.1 | pass | pass
.color-blue1.color-hover:hover | 000 | 29c5ff | 10.5 | pass | pass
.color-blue2 | fff | 1297e0 | 1.22 | fail | fail
.color-blue2.color-hover:hover | fff | 0287d0 | 1.01 | fail | fail
.color-blue3 | fff | 0057a0 | 1.85 | fail | fail
.color-blue3.color-hover:hover | fff | 004790 | 2.30 | fail | fail
.color-turquoise1 | 000 | 5efaf7 | 16.4 | pass | pass
.color-turquoise1.color-hover:hover | 000 | 51f5ea | 15.5 | pass | pass
.color-turquoise2 | fff | 27cbc0 | 1.95 | fail | fail
.color-turquoise2.color-hover:hover | fff | 17bbb0 | 1.64 | fail | fail
.color-turquoise3 | fff | 008b80 | 1.06 | fail | fail
.color-turquoise3.color-hover:hover | fff | 007b70 | 1.30 | fail | fail
.color-green1 | 000 | 8effc1 | 17.2 | pass | pass
.color-green1.color-hover:hover | 000 | 5efca1 | 15.9 | pass | pass
.color-green2 | fff | 2ecc71 | 1.87 | fail | fail
.color-green2.color-hover:hover | fff | 1ebc61 | 1.58 | fail | fail
.color-green3 | fff | 008c31 | 1.10 | fail | fail
.color-green3.color-hover:hover | fff | 007c21 | 1.36 | fail | fail
.color-yellow1 | 000 | fde3a7 | 16.7 | pass | pass
.color-yellow1.color-hover:hover | 000 | ffcf4b | 14.2 | pass | pass
.color-yellow2 | fff | f5ab35 | 2.02 | fail | fail
.color-yellow2.color-hover:hover | fff | f39c12 | 1.79 | fail | fail
.color-yellow3 | fff | d87400 | 1.20 | fail | fail
.color-yellow3.color-hover:hover | fff | c86400 | 1.00 | fail | fail
.color-orange1 | 000 | ffdcb5 | 16.1 | pass | pass
.color-orange1.color-hover:hover | 000 | ffc29b | 13.4 | pass | pass
.color-orange2 | fff | ff926b | 1.79 | fail | fail
.color-orange2.color-hover:hover | fff | f3825b | 1.53 | fail | fail
.color-orange3 | fff | c3522b | 1.16 | fail | fail
.color-orange3.color-hover:hover | fff | b3421b | 1.43 | fail | fail
.color-brown1 | 000 | f6c4a3 | 13.3 | pass | pass
.color-brown1.color-hover:hover | 000 | eab897 | 11.8 | pass | pass
.color-brown2 | fff | ce9c7b | 1.62 | fail | fail
.color-brown2.color-hover:hover | fff | be8c6b | 1.34 | fail | fail
.color-brown3 | fff | 8e5c3b | 1.42 | fail | fail
.color-brown3.color-hover:hover | fff | 7e4c2b | 1.79 | fail | fail
.color-gray1 | 000 | e0e0e0 | 15.9 | pass | pass
.color-gray1.color-hover:hover | 000 | d0d0d0 | 13.6 | pass | pass
.color-gray2 | fff | 909090 | 1.23 | fail | fail
.color-gray2.color-hover:hover | fff | 808080 | 1.00 | fail | fail
.color-gray3 | fff | 505050 | 2.04 | fail | fail
.color-gray3.color-hover:hover | fff | 404040 | 2.62 | fail | fail

For reference here's the script I used to parse the table above:

```python
import json
import cssutils
from urllib.request import urlopen

with open('apps/prairielearn/public/stylesheets/colors.css', 'r') as f:
    css = f.read()
    sheet = cssutils.parseString(css)

print("Class name | Foreground | Background | Ratio | AA | AAA")
print("--- | --- | --- | --- | --- | ---")

for rule in sheet:
    if rule.type == rule.STYLE_RULE:
        foreground = rule.style.getPropertyValue('color').replace('#', '')
        background = rule.style.getPropertyValue('background-color').replace('#', '')
        with urlopen(f'https://webaim.org/resources/contrastchecker/?fcolor={foreground}&bcolor={background}&api') as response:
            result = response.read()
        parsed = json.loads(result)

        print(f"{rule.selectorText} | {foreground} | {background} | {parsed['ratio']} | {parsed['AA']} | {parsed['AAA']}")
```
